### ISSUE TITLE
Improvement: combine list entries when parsing item/struct yaml 

### ISSUE THREAD

**Issue by onkelandy** [smarthomeNG/smarthome#317]
I would recommend that attributes consisting of lists are (at least on demand) combined throughout all items/*.yaml and etc/struct.yaml files.

Assume multiple struct definitions of a plugin:
```
item_structs:
  struct1:
    eval_trigger:
        - item1

  struct2:
    eval_trigger:
        - item2
        - item3
```

When including theses structs in an item using:
```
struct:
    - plugin.struct1
    - plugin.struct2
```
it would be great if the resulting eval_trigger attribute included all list entries from both structs with the following result:
```
    eval_trigger:
        - item1
        - item2
        - item3
```

Same applies for general items that are re-configured in the item tree I guess. Optionally the list entry could get an additional info like ", append" though that might make things even more complicated ;)

**Comment by msinn** [comment]
This is a neat idea. It may make things more complicated to understand for the average user though:

At the moment: When defining an item several times and redefines attributes, the redefined attribute definitions are overwritten by the last definition. This is the case for all attributes of an item.

You propose an exception to the general rule for the attribute *eval_trigger*. If I get you right, you propose the change not only for structs, but for regular item definitions as well?

**Comment by onkelandy** [comment]
Let's put it like this: for structs it seems almost "necessary" to easily combine multiple templates. Right now I figured the main issue is the eval_trigger (for the state engine plugin). But in general I guess any attribute with a list could/should be extended by another list

If this behaviour is only useful for structs or for general yaml definitions needs to be discussed. As well as if combining lists is a standard feature or if it should only take place with additional options activated like..

```
XYZ_combine: true
XYZ:
 - a
 - b

XYZ:
 - c
 - d
```
results in XYZ = [a, b, c, d]

Whereas without the "combine" attribute it would be [c, d]. Of course it would be very useful if there was an entry in the startup log as soon as there are two items with the same name.

**Comment by msinn** [comment]
Is implemented in release 1.7
